ambry-server/src/main/java/com/github/ambry/server/mysql/AccountReportsDao.java

```diff
@@ -0,0 +1,117 @@
+  public static final String PARTITION_ID_COLUMN = "partitionId";
+  public static final String ACCOUNT_ID_COLUMN = "accountId";
+  public static final String CONTAINER_ID_COLUMN = "containerId";
+  public static final String STORAGE_USAGE_COLUMN = "storageUsage";
+  public static final String UPDATED_AT_COLUMN = "updatedAt";
+
+  private static final Logger logger = LoggerFactory.getLogger(AccountReportsDao.class);
+  private static final String insertSql =
+      String.format("INSERT INTO %s (%s, %s, %s, %s, %s, %s, %s) VALUES (?, ?, ?, ?, ?, ?, NOW())",
+          ACCOUNT_REPORTS_TABLE, CLUSTERNAME_COLUMN, HOSTNAME_COLUMN, PARTITION_ID_COLUMN, ACCOUNT_ID_COLUMN,
+          CONTAINER_ID_COLUMN, STORAGE_USAGE_COLUMN, UPDATED_AT_COLUMN);
+  private static final String querySqlForClusterAndHost =
+      String.format("SELECT %s, %s, %s, %s FROM %s WHERE %s = ? AND %s = ?", PARTITION_ID_COLUMN, ACCOUNT_ID_COLUMN,
+          CONTAINER_ID_COLUMN, STORAGE_USAGE_COLUMN, ACCOUNT_REPORTS_TABLE, CLUSTERNAME_COLUMN, HOSTNAME_COLUMN);
+  private final MySqlDataAccessor dataAccessor;
+  private final String clustername;
+  private final String hostname;
+
+  /**
+   * Constructor to create a {@link AccountReportsDao}.
+   * @param dataAccessor The underlying {@link MySqlDataAccessor}.
+   * @param clustername The name of the cluster this host is belonging to, like Ambry-prod, Ambry-video.
+   * @param hostname The name of the host. It should include the hostname and the port.
+   */
+  public AccountReportsDao(MySqlDataAccessor dataAccessor, String clustername, String hostname) {
+    this.dataAccessor = Objects.requireNonNull(dataAccessor, "MySqlDataAccessor is empty");
+    this.clustername = Objects.requireNonNull(clustername, "clustername is empty");
+    this.hostname = Objects.requireNonNull(hostname, "hostname is empty");
+  }
+
+  /**
+   * Update the storage usage for the given account/container.
+   * @param partitionId The partition id of this account/container usage.
+   * @param accountId The account id.
+   * @param containerId The container id.
+   * @param storageUsage The storage usage in bytes.
+   */
+  public void updateStorageUsage(short partitionId, short accountId, short containerId, long storageUsage) {
+    try {
+      long startTimeMs = System.currentTimeMillis();
+      PreparedStatement insertStatement = dataAccessor.getPreparedStatement(insertSql, true);
+      insertStatement.setString(1, clustername);
+      insertStatement.setString(2, hostname);
+      // The data type of partition id, account id and container id are not SMALLINT, but INT in MySQL, for
+      // future extension
+      insertStatement.setInt(3, partitionId);
+      insertStatement.setInt(4, accountId);
+      insertStatement.setInt(5, containerId);
+      insertStatement.setLong(6, storageUsage);
+      insertStatement.executeUpdate();
+      dataAccessor.onSuccess(Write, System.currentTimeMillis() - startTimeMs);
+    } catch (SQLException e) {
```

Comment:
It looks suspicious that the method is eating the exception just so it can be called as a function.  That also raises the question: what if the mysql instance is down during the whole report run - will the report complete successfully even though every sql update failed and logged an error?

Reply:
It will and that's by design. Reporting container storage usage will be done periodically in every 10 minutes. If the writes failed, it's fine to move on and wait for next reports. Besides, reporting container usage is not an essential components for ambry, ambry can survive without it.

Reply:
The part about the report makes sense.  But that knowledge shouldn't be hard wired into the DAO method.  I'd prefer to have it throw SQLException, add a wrapper method in AccountStatsMySqlStore to call it and suppress the exception, and use the wrapper method as your lambda.

Reply:
updated as commented.